lib/action_subscriber/threadpool.rb

```diff
@@ -4,17 +4,29 @@ class Threadpool
     # Class Methods
     #
     def self.busy?
-      (pool.pool_size == pool.busy_size)
+      pools.any? do |_pool_name, pool|
+        pool.pool_size == pool.busy_size
+      end
     end
 
-    def self.perform_async(*args)
-      self.pool.async.perform(*args)
+    def self.new_pool(name, pool_size = nil)
+      fail ArgumentError, "#{name} already exists as a threadpool" if pools.key?(name)
+      pool_size ||= ::ActionSubscriber.config.threadpool_size
+      pools[name] = ::Lifeguard::InfiniteThreadpool.new(
+        :pool_size => pool_size
+      )
     end
 
-    def self.pool
-      @pool ||= ::Lifeguard::InfiniteThreadpool.new(
-        :pool_size => ::ActionSubscriber.config.threadpool_size
-      )
+    def self.pool(which_pool = :default)
+      pools[which_pool]
+    end
```

Comment:
What do you think about the name `current_pool` instead of `which_pool`?  

`which_pool` kind of confused me when I was reading through this.